frontend/app/work_packages/activities/user-activity-directive.js

```diff
@@ -99,20 +104,37 @@ module.exports = function($uiViewScroll,
       };
 
       scope.quoteComment = function() {
-        exclusiveEditController.setQuoted(quotedText(scope.activity.props.comment.raw));
-        var elem = angular.element('#' + scope.inputElementId);
-        $uiViewScroll(elem);
-        elem.focus();
+        scope.$emit(
+          'workPackage.comment.quoteThis',
+          quotedText(scope.activity.props.comment.raw)
+        );
       };
 
-      scope.updateComment = function(comment) {
-        var comment = angular.element('#edit-comment-text').val();
+      scope.updateComment = function() {
+        var comment = element.find('.edit-comment-text').val();
         ActivityService.updateComment(scope.activity, comment).then(function(activity){
           scope.$emit('workPackageRefreshRequired', '');
           scope.inEdit = false;
         });
       };
 
+      scope.toggleCommentPreview = function() {
+        scope.inPreview = !scope.inPreview;
+        scope.previewHtml = '';
+        if (scope.inPreview) {
+          TextileService.renderWithWorkPackageContext(
+            EditableFieldsState.workPackage.form,
+            scope.activity.props.comment.raw
+          ).then(function(r) {
+            scope.previewHtml = $sce.trustAsHtml(r.data);
+
+
+          }), function() {
+            this.inPreview = false;
+          };
```

Comment:
I believe that houndci spotted an error here. Instead of having written
```
then(function(r) {
...
}, function() { 
...
})
```
where the second function becomes the `reject` callback of the promise, you have misplaced the closing bracket which will render the intended `reject` callback useless.